### ISSUE TITLE
[Bug]: docker build fails on compile step

### ISSUE THREAD

**Issue by BlitzkriegSoftware** [a2aproject/a2a-inspector#103]
### What happened?

`docker build --progress=plain -t a2a-inspector . > build.log 2>&1`

[build.log](https://github.com/user-attachments/files/22936640/build.log)

### Relevant log output

```shell
-- SNIP -- (See log file for full details)
#21 5.964       -c src/c/_cffi_backend.c -o
#21 5.964       build/temp.linux-x86_64-cpython-314/src/c/_cffi_backend.o
#21 5.964 
#21 5.964       [stderr]
#21 5.964 
#21 5.964           No working compiler found, or bogus compiler options passed to
#21 5.964           the compiler from Python's standard "distutils" module.  See
#21 5.964           the error messages above.  Likely, the problem is not related
#21 5.964           to CFFI but generic to the setup.py of any Python package that
#21 5.964           tries to compile C code.  (Hints: on OS/X 10.8, for errors about
#21 5.964           -mno-fused-madd see http://stackoverflow.com/questions/22313407/
#21 5.964           Otherwise, see https://wiki.python.org/moin/CompLangPython or
#21 5.964           the IRC channel #python on irc.libera.chat.)
#21 5.964 
#21 5.964           Trying to continue anyway.  If you are trying to install CFFI from
#21 5.964           a build done in a different context, you can ignore this warning.
#21 5.964       
#21 5.964       /tmp/.tmpwCXAyb/builds-v0/.tmpxIuwIS/lib/python3.14/site-packages/setuptools/dist.py:759:
#21 5.964       SetuptoolsDeprecationWarning: License classifiers are deprecated.
#21 5.964       !!
#21 5.964 
#21 5.964       
#21 5.964       ********************************************************************************
#21 5.964               Please consider removing the following classifiers in favor of a
#21 5.964       SPDX license expression:
#21 5.964 
#21 5.964               License :: OSI Approved :: MIT License
#21 5.964 
#21 5.964               See
#21 5.964       https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license
#21 5.964       for details.
#21 5.964       
#21 5.964       ********************************************************************************
#21 5.964 
#21 5.964       !!
#21 5.964         self._finalize_license_expression()
#21 5.964       error: command 'cc' failed: No such file or directory
#21 5.964 
#21 5.964       hint: This usually indicates a problem with the package or the build
#21 5.964       environment.
#21 5.964   help: `cffi` (v1.17.1) was included because `a2a-inspector-workspace`
#21 5.964         (v0.1.0) depends on `jwcrypto` (v1.5.6) which depends on
#21 5.964         `cryptography` (v45.0.4) which depends on `cffi`
#21 ERROR: process "/bin/sh -c uv sync --no-cache" did not complete successfully: exit code: 1
------
 > [stage-1  5/10] RUN uv sync --no-cache:
5.964 
5.964       !!
5.964         self._finalize_license_expression()
5.964       error: command 'cc' failed: No such file or directory
5.964 
5.964       hint: This usually indicates a problem with the package or the build
5.964       environment.
5.964   help: `cffi` (v1.17.1) was included because `a2a-inspector-workspace`
5.964         (v0.1.0) depends on `jwcrypto` (v1.5.6) which depends on
5.964         `cryptography` (v45.0.4) which depends on `cffi`
------
Dockerfile:27
--------------------
  25 |     RUN pip install uv
  26 |     COPY pyproject.toml uv.lock ./
  27 | >>> RUN uv sync --no-cache
  28 |     RUN uv pip install validators
  29 |     COPY backend/ ./backend/
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c uv sync --no-cache" did not complete successfully: exit code: 1
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by pjmagee** [comment]
Reproduced this bug as well myself. cloned the repo, tried to follow instructions.
